src/components/ArticleComponents/Codebox/index.tsx

```diff
@@ -20,49 +20,66 @@ const replaceLanguages = (language: string) =>
     .replace(/mjs|cjs|javascript/i, 'js')
     .replace(/console|shell/i, 'bash');
 
-const replaceLabelLanguages = (language: string) =>
-  language
-    .replace(/javascript/i, 'cjs')
-    .replace(/console|shell/i, 'bash')
-    .toUpperCase();
-
 const Codebox = ({ children: { props } }: Props): JSX.Element => {
   const [parsedCode, setParsedCode] = useState('');
   const [copied, copyText] = useCopyToClipboard();
+  const [langIndex, setLangIndex] = useState(0);
+  const [labelLang, setLabelLang] = useState('LANG');
+  const [stringCode, setStringCode] = useState('');
 
   // eslint-disable-next-line react/prop-types
-  const className = props.className || 'text';
-
+  const className = props.className || 'LANG';
   // Language Matches in class
   const matches = className.match(/language-(?<lang>.*)/);
-
   // Language name from classname
   const language = matches?.groups?.lang || 'text';
+  const LabelArray = language.split('|') || ['LANG'];
+  const codeArray = props.children
+    ? props.children.toString().split('------\n')
+    : [''];
 
-  // Actual Code into a stringified format
-  const stringCode = props.children?.toString() || '';
+  const setLabel = () => {
+    setLabelLang(LabelArray[langIndex]);
+  };
+
+  const setCode = () => {
+    setStringCode(codeArray[langIndex]);
+    setParsedCode(
+      sanitize(
+        highlight(codeArray[langIndex], languages.javascript, 'javascript')
+      )
+    );
+  };
+
+  const handleLangChange = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setLangIndex((langIndex + 1) % LabelArray.length);
```

Comment:
I don't think this logic is correct.

Reply:
There are a few issues with this approach (and logic).

- We want to be able to switch to a specific language (index); the current logic tries to always add + 1 to the current index, which is broken. It works, and it even works if you have more than 2 items in the array. It will keep summing + 1 until it reaches the length, and then it goes back to 0. So the suggested idea works, **but** the big caveat is that it doesn't solve the underlying issue.
- For that, you would want X amount of buttons for each language.
    ```jsx
    {languageOptions.map((language, index) => 
        <button key={index} onClick={setLanguageIndex(index)}>
            {replaceLanguageCode(language)}
        </button>
    )}
  ```
  This code is just an example of how it should be done (IMHO).
- There's also the issue of how you're handling code change. You've created `setLabel` and `setCode`, an anti-pattern for the proposed solution.
    You should update the existing `useEffect` to do:
    ```jsx
    useEffect(() => {
		// Keep in mind that `const language = languageOptions[langIndex];`
    	const parsedLanguage = replaceLanguages(language);
    	const prismLanguage = languages[parsedLanguage] || languages.text;

    	setParsedCode(
            sanitize(highlight(stringCode, prismLanguage, parsedLanguage))
    	);
    }, [langIndex]);
	```
- Remember that the examples/snippets above are basic/fundamental React.js concepts. And how much time I'm spending here to teach these concepts. That's part of my message that I tried to explain before. Not trying to be rude, but just saying even tho I personally don't mind teaching this and explaining these concepts to you, I'm in no position here as a mentor/teacher, and I don't have the time right now to explain these concepts every single time 😅